cmd/consumer/consumer.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/Shopify/sarama"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+
+	"github.com/vmware/go-ipfix/pkg/kafka/consumer"
+	"github.com/vmware/go-ipfix/pkg/kafka/consumer/protobuf"
+)
+
+const (
+	logToStdErrFlag = "logtostderr"
+)
+
+var (
+	Brokers           string
+	MsgDelimitWithLen bool
+	KafkaTopic        string
+)
+
+func initLoggingToFile(fs *pflag.FlagSet) {
+	var err error
+
+	_, err = fs.GetBool(logToStdErrFlag)
+	if err != nil {
+		// Should not happen. Return for safety.
+		return
+	}
+}
+
+func addKafkaConsumerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&Brokers, "brokers", "kafka-service:9092", "Kafka broker addresses delimited by comma")
+	fs.BoolVar(&MsgDelimitWithLen, "consumer.delimitlen", false, "Kafka consumer message delimited with length")
+	fs.StringVar(&KafkaTopic, "consumer.topic", "AntreaTopic", "Kafka topic")
+}
+
+func signalHandler(stopCh chan struct{}, consumer *consumer.KafkaConsumer) {
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+	for {
+		select {
+		case msg := <-consumer.GetMsgChan():
+			consumer.DecodeAndPrintMsg(msg)
+		case <-signalCh:
+			close(stopCh)
+			return
+		}
+	}
+}
+
+func run() error {
+	klog.Info("Starting Kafka consumer")
+	flowType := &protobuf.AntreaFlowMsg{}
```

Comment:
this is ok for now.. we should consider an interface that takes proto schema name as input in the future.

Reply:
Do you mean similar to what producer does right now?

Reply:
I meant the user of kafka producer can take in multiple proto schema as input and create producer with appropriate convertor. Consumer application should take the proto schema name as input and decode based on that. We are currently supporting only Antrea proto schema.